File: lib_common/src/main/java/com/fltry/module/lib_common/BaseActivity.java
package com.fltry.module.lib_common;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.widget.TextView;

/**
 * Created by tol on 2018/3/21.
 * base类
 */
public abstract class BaseActivity extends AppCompatActivity {

    //登录返回的token
    public static String JWT = "";

    public Context mContext;
    private TextView mToolbarTitle;
    private Toolbar mToolbar;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mContext = this;

        setContentView(getLayoutId());

        mToolbar = findViewById(R.id.toolbar);
        mToolbarTitle = findViewById(R.id.toolbar_title);
        if (mToolbar != null) {
            //将Toolbar显示到界面
            setSupportActionBar(mToolbar);
        }

        if (mToolbarTitle != null) {
            //设置默认的标题不显示
            getSupportActionBar().setDisplayShowTitleEnabled(false);
        }
    }



    @Override
    protected void onStart() {
        super.onStart();
        /**
         * 判断是否有Toolbar,并默认显示返回按钮
         */
        if (null != getToolbar() && isShowBacking()) {
            showBack();
        }
    }

    /**
     * 设置头部标题
     *
     * @param title
     */
    public void setToolBarTitle(CharSequence title) {
        if (mToolbarTitle != null) {
            mToolbarTitle.setText(title);
        } else {
            getToolbar().setTitle(title);
            setSupportActionBar(getToolbar());
        }
    }

    /**
     * 获取头部标题的TextView
     *
     * @return
     */
    public TextView getToolbarTitle() {
        return mToolbarTitle;
    }

    /**
     * this Activity of tool bar.
     * 获取头部.
     *
     * @return support.v7.widget.Toolbar.
     */
    public Toolbar getToolbar() {
        return (Toolbar) findViewById(R.id.toolbar);
    }

    /**
     * 版本号小于21的后退按钮图片
     */
    private void showBack() {
        //setNavigationIcon必须在setSupportActionBar(toolbar);方法后面加入
        getToolbar().setNavigationIcon(R.mipmap.fhj_hs);
        getToolbar().setNavigationOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                onBackPressed();
            }
        });
    }

    /**
     * 是否显示后退按钮,默认显示,可在子类重写该方法.
     *
     * @return
     */
    protected boolean isShowBacking() {
        return true;
    }

    /**
     * this activity layout res
     * 设置layout布局,在子类重写该方法.
     *
     * @return res layout xml id
     */
    protected abstract int getLayoutId();
}

File: app/src/main/java/com/fltry/androidlibs/ui/BaseActivity.kt
package com.fltry.androidlibs.ui

import android.content.Context
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.Toolbar
import android.view.View
import android.widget.TextView
import butterknife.ButterKnife
import com.fltry.androidlibs.R

/**
 * Created by tol on 2018/3/21.
 * base类
 */
abstract class BaseActivity : AppCompatActivity() {

    lateinit  var mContext: Context
    /**
     * 获取头部标题的TextView
     *
     * @return
     */
    var toolbarTitle: TextView? = null
        private set
    private var mToolbar: Toolbar? = null

    /**
     * this Activity of tool bar.
     * 获取头部.
     *
     * @return support.v7.widget.Toolbar.
     */
    val toolbar: Toolbar?
        get() = findViewById<View>(R.id.toolbar) as Toolbar

    /**
     * 是否显示后退按钮,默认显示,可在子类重写该方法.
     *
     * @return
     */
    protected val isShowBacking: Boolean
        get() = true

    /**
     * this activity layout res
     * 设置layout布局,在子类重写该方法.
     *
     * @return res layout xml id
     */
    protected abstract val layoutId: Int


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mContext = this
        setContentView(layoutId)

        mToolbar = findViewById(R.id.toolbar)
        toolbarTitle = findViewById(R.id.toolbar_title)
        if (mToolbar != null) {
            //将Toolbar显示到界面
            setSupportActionBar(mToolbar)
        }
        if (toolbarTitle != null) {
            //getTitle()的值是activity的android:lable属性值
            toolbarTitle!!.text = title
            //设置默认的标题不显示
            supportActionBar!!.setDisplayShowTitleEnabled(false)
        }
        initSdk();
    }

    private fun initSdk() {
        ButterKnife.bind(this)
    }

    override fun onStart() {
        super.onStart()
        /**
         * 判断是否有Toolbar,并默认显示返回按钮
         */
        if (null != toolbar && isShowBacking) {
            showBack()
        }
    }

    /**
     * 设置头部标题
     *
     * @param title
     */
    fun setToolBarTitle(title: CharSequence) {
        if (toolbarTitle != null) {
            toolbarTitle!!.text = title
        } else {
            toolbar!!.title = title
            setSupportActionBar(toolbar)
        }
    }

    /**
     * 版本号小于21的后退按钮图片
     */
    private fun showBack() {
        //setNavigationIcon必须在setSupportActionBar(toolbar);方法后面加入
        toolbar!!.setNavigationIcon(R.mipmap.fhj_hs)
        toolbar!!.setNavigationOnClickListener { onBackPressed() }
    }
}
